src/main/java/org/wildfly/naming/client/util/NamingUtils.java

```diff
@@ -57,7 +57,7 @@ public static CompositeName toCompositeName(final Name name) throws InvalidNameE
             return (CompositeName) name;
         } else {
             final CompositeName compositeName = new CompositeName();
-            compositeName.add(name.toString());
+            compositeName.addAll(new CompositeName(name.toString()));
```

Comment:
What I'm striving to arrive at is a way for the protocol to be "right" (i.e. passing a composite name always, with the common case having the first segment be a SimpleName) without having to rewrite large chunks of the WildFly naming infrastructure (yet) as we can/should do this in a later phase or release.